### DISCUSSION TITLE
Phone number validation using a custom validator

### DISCUSSION THREAD

**Discussion by bharatmca86** [json-schema-org/community#582] [General]
Is it possible to.write a custom validator to validate the international phone numbers using Google's libphonenumber library along with some regex  ?

**Comment by gregsdennis** [comment]
It should be possible, sure.

If you want to use the regex approach, then the `pattern` keyword is your friend.

If you want to use the library directly, you'll need to either define a custom keyword or a custom value for `format`, which means you'll be implementing the validation code yourself.  Whether those approaches are supported by the JSON Schema library you're using (and how to do it) will depend on the library, and you'll need to ask them.  Additionally, you'll need to have that custom logic _everywhere_ that the schema is used to ensure interoperability.
